data/shared/citizen/scripting/lua/scheduler.lua

```diff
@@ -21,17 +27,32 @@ CreateThread = Citizen.CreateThread
 
 function Citizen.CreateThreadNow(threadFunction)
 	local coro = coroutine.create(threadFunction)
+	local callerInfo = debug.getinfo(3, 'Sl')
+
+	if not callerInfo then
+		callerInfo = debug.getinfo(2, 'Sl')
+	end
 
 	local t = {
 		coroutine = coro,
-		wakeTime = 0
+		wakeTime = 0,
+		resource = GetCurrentResourceName(),
+		file = callerInfo.short_src,
+		line = callerInfo.currentline,
+		wait = 0
 	}
 
 	-- add new thread and save old thread
 	local oldThread = curThread
 	curThread = t
 
+	local startAt = GetGameTimer()
 	local result, err = coroutine.resume(coro)
+	local time = GetGameTimer() - startAt
+
+	if time > 10 then
```

Comment:
Not sure about this limit

Reply:
`>= 150` for consistency with the underneath hitch warning?

Reply:
Not really a hitch warning is for the whole frame where there can be many threads executed here

Reply:
IMO something between 50 and 100 would be fine, 10ms is way too low